src/main/java/me/zeroeightsix/kami/module/modules/movement/AutoWalk.java

```diff
@@ -33,25 +34,63 @@
             case BACKWARDS:
                 event.getMovementInput().moveForward = -1;
                 break;
-            case PATH:
-                if (Pathfind.points.isEmpty()) return;
-                event.getMovementInput().moveForward = 1;
-                if (mc.player.isInWater() || mc.player.isInLava()) mc.player.movementInput.jump = true;
-                else if (mc.player.collidedHorizontally && mc.player.onGround) mc.player.jump();
-                if (!MODULE_MANAGER.isModuleEnabled(Pathfind.class) || Pathfind.points.isEmpty()) return;
-                PathPoint next = Pathfind.points.get(0);
-                lookAt(next);
-                break;
         }
     });
 
-    private void lookAt(PathPoint pathPoint) {
-        double[] v = calculateLookAt(pathPoint.x + .5f, pathPoint.y, pathPoint.z + .5f, mc.player);
-        mc.player.rotationYaw = (float) v[0];
-        mc.player.rotationPitch = (float) v[1];
+    @EventHandler
+    public Listener<ServerDisconnectedEvent> kickListener = new Listener<>(event -> {
+        if(mode.getValue().equals(AutoWalkMode.BARITONE) && isEnabled()) {
```

Comment:
Formatting, missed a space

Reply:
Also maybe make disabling optional and just have it return, so it will continue autowalking when you reconnect, which is a MAJOR complaint in Impact / Baritone

Reply:
NullPointerExceptions everywhere, it's hella unstable, I tried doing that during its development. There's a reason why that's not in Impact.

Reply:
try 
catch

Reply:
i've always coded on the principle of never ignoring exceptions, but i'll give that a go anyway i guess

Reply:
Tried another way, and it didn't work. You MUST disable the baritone process, cos when you don't, even if you relaunch the game, when you enable and try to disable autowalk, it keeps executing the old process, for some reason and so you have to cancel manually.

Reply:
I see.